tox.ini

```diff
@@ -170,19 +170,21 @@ commands =
 commands =
     docker build -t certbot-compatibility-test -f certbot-compatibility-test/Dockerfile .
     docker build -t apache-compat -f certbot-compatibility-test/Dockerfile-apache .
-    docker run --rm -it apache-compat -c apache.tar.gz -vvvv
+    script -qec "docker run --rm -it apache-compat -c apache.tar.gz -vvvv"
```

Comment:
Ah! This one is tricky. I would say even that is it a master of trickery, or a trickyception.

First, we have several tests, `certbot-auto` tests in particular, where we check the behavior of `certbot-auto` with and without interactive terminal. We do some tricks in these tests to make `certbot-auto` believe it is running or not in an interactive terminal. This is the first level of tricks.

But to make this work, the tests scripts must start in an interactive terminal. Since there are inside dockers, we use `docker run -it` to make the script thinks it is in an interactive terminal (while a simple `docker run` would make the execution non interactive). This is the second level of tricks.

However, if the place where `docker run -it` is not a proper terminal, you will receive a `the input device is not a TTY` error while trying to execute it. For instance, try `echo "Hello" | docker run -it --rm ubuntu:18.04 echo World!`.

This what is happening in Azure Pipeline, which is executing the bash commands in a non-interactive mode. Which makes sense, since you cannot input things in a CI job. Travis was simulating in some way a terminal for these commands, so it was working.

This is the third level of tricks. Initially I used `unbuffer`, from the `expect` package, and used it for Linux jobs directly to invoke the `tox` command. So the "Run tox" step in `tox-steps.yml` was actually invoking `unbuffer tox` when `unbuffer` is available (and that is why `expect` was installed, as noticed above). Indeed `unbuffer` call has the side effect to make the binary/script believe it is running in a interactive mode. At least for `docker run -it`. This corresponds to the run https://dev.azure.com/certbot/certbot/_build/results?buildId=2064&view=results.

However I thought it was not elegant, because it was specific to Azure Pipeline, to fixing the behavior in other situations where `tox` would be called for these tests in a non-interactive mode, and because the trick was a side effect of `unbuffer`. Then I found `script` which has this side effect as an "official" one, describe it its documentation. As an additional benefit `script` comes from `util-linux`, which is an almost always installed package in a strandard Linux distribution. 

So I wanted to use it one or two level below, to make all calls benefit from it, not the specific situation of Azure. Sadly the best place, which is directly inside the `CMD` instruction of the Dockerfile, is not possible because CentOS6 does not have `util-linux`, but only `util-linux-ng` which install the BSD version, with less possibilities (in particular, transfering the exit code of the callee). So I put it one layer up, in `tox.ini`, so any call of `tox` will continue to benefit from it. As a side note, `unbuffer` can be installed in CentOS 6, but sadly (again) is not sufficient to trick the `certbot-auto` scripts ...

However, considering that it adds some complexity, and break for MacOS (with no straightforward solution instead a manual script implementation), nobody ever complained about it, I think I will just keep things simple, and call `script` from `tox-steps.yml`.

What do you think @bmw?

Reply:
What about just removing the `-i` argument from the Docker commands? I tested this approach after making the changes in [this commit](https://github.com/certbot/certbot/commit/26bab1ae33839b5fbb773c9ea17fd355a0743306) and it seemed to work both locally and in Azure.

Reply:
You can see all tests passing on this commit at https://dev.azure.com/certbot/certbot/_build/results?buildId=2180&view=results.

Reply:
Or ... simply that, indeed. I am so used to put -i and -t together that I did not even though that having the TTY is just sufficient for `certbot-auto`, which tests that a terminal is attached, not that `/dev/stdin` is open or closed ...

Reply:
Fixed.